hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/MergeOnReadIncrementalRelation.scala

```diff
@@ -95,6 +108,9 @@ class MergeOnReadIncrementalRelation(sqlContext: SQLContext,
       hadoopConf = new Configuration(conf)
     )
 
+    // setUp tableRequiredSchema
+    tableRequiredSchema = requiredSchema.structTypeSchema
```

Comment:
Why do we need this one? Can we use `requiredSchema` directly?

Reply:
> @xiarixiaoyao while i understand your intent that we can speed things up for some of the queries, i don't think enabling vectorized reader for Incremental queries will bring an universal speed up for all queries -- vectorized reader would mean that we first fetch records and only do filtering in memory, which i could clearly could be a disadvantage for large tables.
> 
> So my suggestion would be to NOT enable vectorized-reader for Incremental queries by default, but instead let individual users to decide on whether they want to enable it or not.

@alexeykudinkin  I don't think record level filter has better performance. Spark is off by default for this parameter. 
Read out the data, vectorize it, and then filter it better. I will take out 100g data to give the test results

Reply:
@xiarixiaoyao sorry for the late reply.

I understand that it reads it by batches, but my point still stands -- unfortunately, (i don't really understand why those 2 things are incompatible, but nevertheless) when we use vectorization -- Parquet filter push-down stops working, which means that now every row-group would be read in memory, before being filtered out as irrelevant. 

For use-cases with localized storage (ie adjacent to the Compute) it might not be a problem, but it's more likely to be a problem for use-cases with disaggregated Compute and Storage (for ex, in Cloud) -- now we will have to read whole row-group for Spark to be able to filter it out in memory.

I'd suggest additionally to the Benchmark that you ran locally, can we also test similar setup in Cloud (like AWS, or wherever), so that we get a better sense what the impact would be in Cloud env?